src/celeritas/neutron/interactor/NeutronElasticInteractor.hh

```diff
@@ -0,0 +1,155 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/interactor/NeutronElasticInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "celeritas/Constants.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/neutron/data/NeutronElasticData.hh"
+#include "celeritas/phys/FourVector.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+#include "detail/InvariantQ2Sampler.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform neutron elastic scattering based on the CHIPS (Chiral invariant
+ * phase space) model.
+ *
+ * \note This performs the sampling procedure as in G4HadronElastic,
+ * G4ChipsElasticModel and G4ChipsNeutronElasticXS, as partly documented
+ * in section 21.1.3 of the Geant4 Physics Reference (release 11.2).
+ */
+class NeutronElasticInteractor
+{
+    //!@{
+    //! \name Type aliases
+    using Energy = units::MevEnergy;
+    using Mass = units::MevMass;
+    //!@}
+
+  public:
+    // Construct from shared and state data
+    inline CELER_FUNCTION
+    NeutronElasticInteractor(NeutronElasticRef const& shared,
+                             ParticleTrackView const& particle,
+                             Real3 const& inc_direction,
+                             IsotopeView const& target);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// TYPES ////
+
+    using UniformRealDist = UniformRealDistribution<real_type>;
+
+    //// DATA ////
+
+    // Constant shared data
+    NeutronElasticRef const& shared_;
+    // Incident neutron energy
+    units::MevEnergy const inc_energy_;
+    // Incident neutron direction
+    Real3 const& inc_direction_;
+    // Target nucleus
+    IsotopeView const& target_;
+
+    // Value of neutron MeVMass
+    real_type neutron_mass_;
+    real_type neutron_p_;
+
+    // Sampler
+    UniformRealDist sample_phi_;
+    detail::InvariantQ2Sampler sample_Q2_;
```

Comment:
Can you rename `2 -> _sq`  everwhere? And also please do not use capital letters in variable names.

Reply:
This was also changed in the earlier commit - changed `InvariantQ2Sampler` to `MomentumTransferSampler` and `x2` to `x_sq` in relevant places.